### PR TITLE
PHP-1505: Use long and size_t for GridFS file lengths

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHP-1505

Please run `fallocate -l 5G /tmp/5gb` to create a file for the regression test.

### CODE REVIEW

File: gridfs/gridfs.c [hunk 1]
```diff
@@ -550,7 +550,7 @@ PHP_METHOD(MongoGridFS, storeBytes)
  * - upload date
  * - length
  * these fields are only added if the user hasn't defined them. */
-static int setup_file_fields(zval *zfile, char *filename, int length TSRMLS_DC)
+static int setup_file_fields(zval *zfile, char *filename, long length TSRMLS_DC)
```
Comment (derickr) [inline gridfs/gridfs.c]: Why not size_t here?

File: gridfs/gridfs.c [hunk 2]
```diff
@@ -444,7 +444,7 @@ PHP_METHOD(MongoGridFS, storeBytes)
 
 	/* merge extra & zfile and add _id if needed */
 	zid = setup_extra(zfile, extra TSRMLS_CC);
-	setup_file_fields(zfile, NULL, bytes_len TSRMLS_CC);
+	setup_file_fields(zfile, NULL, (long) bytes_len TSRMLS_CC);
```
Comment (derickr) [inline gridfs/gridfs.c]: I'd change the int bytes_len definition to "size_t bytes_len" too (line 411).

File: gridfs/gridfs.c [hunk 3]
```diff
@@ -70,7 +70,7 @@ typedef struct {
 	struct stat sb;
 } php_stdio_stream_data;
 
-static int setup_file_fields(zval *zfile, char *filename, int size TSRMLS_DC);
+static int setup_file_fields(zval *zfile, char *filename, long size TSRMLS_DC);
```
Comment (derickr) [inline gridfs/gridfs.c]: This can potentially also be a size_t instead. Should be the same as below though.

**Comment by derickr** [comment]
Did you test this with files larger than 4GB as well?

File: gridfs/gridfs.c (hunk 2 above)
Comment (jmikola) [inline gridfs/gridfs.c]: `bytes_len` is an int populated by zpp, so I don't believe it's a candidate for change.

File: gridfs/gridfs.c (hunk 1 above)
Comment (jmikola) [inline gridfs/gridfs.c]: This is passed to `add_assoc_long()`. Also, the value from the `MongoGridFS::storeFile()` code path comes from `setup_file()` and `ftell()`, which returns a `long`. I don't see any reason to switch to `size_t` when both the input and output are already `long`.

File: gridfs/gridfs.c (hunk 3 above)
Comment (jmikola) [inline gridfs/gridfs.c]: See previous comment.

**Comment by jmikola** [comment]
> Did you test this with files larger than 4GB as well?

Manually verified with `fallocate -l 5G /tmp/5gb`, which results in a length field of `int(5368709120)`. :+1:

**Comment by derickr** [comment]
LGTM

**Pull request merged**